File: algorithms/Ransom Note/Ransom Note.js
var canConstruct = function(ransomNote, magazine) {
    map = {};
    for(let i of magazine){
        map[i] = (map[i] || 0) + 1;
    }
    for(let i of ransomNote){
        if(!map[i]){
            return false
        }
        map[i]--;
    }
    return true
};

File: algorithms/Ransom Note/Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        for (int i = 0; i < ransomNote.length(); ++i) {
            String t = ransomNote.substring(i, i+1);
            int idx = magazine.indexOf(t);
            if (idx == -1) return false;
            magazine = magazine.substring(0, idx) + magazine.substring(idx+1);
        }
        return true;
    }
}
